From 9f99ead19791a9b3e76581155a60cb4654aaad76 Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Wed, 24 Jul 2019 19:33:53 -0700
Subject: [PATCH 0645/1852] v4l: xilinx: dma: Copy format resolution to crop
 rectangle

commit 71c88f98bd6d6986cf7c8c540250f33ed10fa46c from
https://github.com/Xilinx/linux-xlnx.git

This patch copies the format resolution to crop rectangle
structure, this will be updated when s_selection is called.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dma.c | 14 ++++++++++++--
 drivers/media/platform/xilinx/xilinx-dma.h |  2 ++
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dma.c b/drivers/media/platform/xilinx/xilinx-dma.c
index bb783d953224..694dd9644de7 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-dma.c
@@ -1107,10 +1107,20 @@ xvip_dma_set_format(struct file *file, void *fh, struct v4l2_format *format)
 	if (vb2_is_busy(&dma->queue))
 		return -EBUSY;
 
-	if (V4L2_TYPE_IS_MULTIPLANAR(dma->format.type))
+	if (V4L2_TYPE_IS_MULTIPLANAR(dma->format.type)) {
 		dma->format.fmt.pix_mp = format->fmt.pix_mp;
-	else
+
+		/*
+		 * Save format resolution in crop rectangle. This will be
+		 * updated when s_slection is called.
+		 */
+		dma->r.width = format->fmt.pix_mp.width;
+		dma->r.height = format->fmt.pix_mp.height;
+	} else {
 		dma->format.fmt.pix = format->fmt.pix;
+		dma->r.width = format->fmt.pix.width;
+		dma->r.height = format->fmt.pix.height;
+	}
 
 	dma->fmtinfo = info;
 
diff --git a/drivers/media/platform/xilinx/xilinx-dma.h b/drivers/media/platform/xilinx/xilinx-dma.h
index 4a5966c4d943..d2e68fde42bf 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.h
+++ b/drivers/media/platform/xilinx/xilinx-dma.h
@@ -63,6 +63,7 @@ static inline struct xvip_pipeline *to_xvip_pipeline(struct media_entity *e)
  * @port: composite device DT node port number for the DMA channel
  * @lock: protects the @format, @fmtinfo and @queue fields
  * @format: active V4L2 pixel format
+ * @r: crop rectangle parameters
  * @fmtinfo: format information corresponding to the active @format
  * @poss_v4l2_fmts: All possible v4l formats supported
  * @poss_v4l2_fmt_cnt: number of supported v4l formats
@@ -90,6 +91,7 @@ struct xvip_dma {
 
 	struct mutex lock;
 	struct v4l2_format format;
+	struct v4l2_rect r;
 	const struct xvip_video_format *fmtinfo;
 	u32 *poss_v4l2_fmts;
 	u32 poss_v4l2_fmt_cnt;
-- 
2.31.1

